src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -177,9 +183,41 @@ private constructor(
     if (messages.lastOrNull()?.id == message.id) {
       messages.removeLast()
     }
-    messages.add(message)
-    chatPanel.addOrUpdateMessage(message)
-    HistoryService.getInstance(project).updateChatMessages(internalId, messages)
+    if (messages.size == 0) {
```

Comment:
Please encapsulate getting the model into separate method (if needed can be just with callback, or it can return `CompletableFuture`)